File: backend/app/models/action_item.py
from sqlalchemy import Column, Integer, String, ForeignKey, Text, Float
from sqlalchemy.orm import relationship
from app.db.base_class import Base

class ActionItem(Base):
    id = Column(Integer, primary_key=True, index=True)
    content = Column(Text, nullable=False)
    type = Column(String) # task, follow-up, decision, question
    timestamp = Column(Float)
    assignee = Column(String)
    status = Column(String, default="pending")
    
    session_id = Column(Integer, ForeignKey("session.id"))
    session = relationship("Session", back_populates="action_items")

File: backend/models/ActionItem.js
module.exports = (sequelize, DataTypes) => {
  return sequelize.define("ActionItem", {
    id: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: true
    },
    content: {
      type: DataTypes.TEXT,
      allowNull: false
    },
    type: DataTypes.STRING,
    timestamp: DataTypes.FLOAT,
    assignee: DataTypes.STRING,
    status: {
      type: DataTypes.STRING,
      defaultValue: "pending"
    },
    session_id: DataTypes.INTEGER
  }, {
    tableName: "action_items",
    timestamps: false
  });
};
